cmd/pack/main_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+func TestRunPack(t *testing.T) {
+	linkerdOut, _ := ioutil.ReadFile("./testdata/linkerd_out.yaml")
```

Comment:
Do you suppose we could get away with a lighter test fixture? Perhaps using afero and yaml consts (as is done in the StackManager tests)?

Are there different conditions this testdata is providing? If not, can we simplify the input/output? If so, can we split those inputs and outputs into easier to reason about yaml blocks, perhaps limiting what is being tested in each case?

Reply:
If the testdata is being reused as example data, I think we should avoid that as it could restrict use for the other case.

Reply:
I think this is certainly a fair idea :) To be clear though, `testdata` is a [special dir](https://dave.cheney.net/2016/05/10/test-fixtures-in-go) that is ignored in go builds, so the only downside here is a larger repo size and arguably less clean tests (depending on your preference). My motivation for using actual configs here was mostly speed (i.e. it was easy to implement quickly) and shows some simple examples for a user of the tool who wouldn't be looking at the code necessarily.

Reply:
I didn't know about the `testdata` directory, the binary size wouldn't be affected as constants in `_test.go` files either. Now knowing that `testdata` carries this predefined meaning I'm less concerned about its introduction here, even with the bulky (fragile?) chart yamls.

Reply:
+1 to this; the test data is so large that it seems like it will be hard to manage. It's also hard to read through. It seems like using a single, smaller example which makes it less intimidating for someone to read the input and expected output would be helpful for maintainability and approachability. It looks like we're already doing that with the namespace example, so I'd consider just reusing that as the single happy path case for this test suite.